source/keymap.js

```diff
@@ -11,8 +11,327 @@
 			413 : 'stop',
 			19  : 'pause',
 			412 : 'rewind',
-			417 : 'fastforward'
+			417 : 'fastforward',
+			461	: 'back'
 		});
 	}
 
-})(enyo, this);
+	enyo.singleton({
+		name: 'moon.History',
+
+		/**
+		* Target object of back key.
+		* It is also the last object which history.pushState() is executed.
+		* When we press back key or call history.back(), it is target object.
+		*
+		* @private
+		*/
+		_currentObj: null,
+
+		/**
+		* Stack of Back key target and handler
+		*
+		* @private
+		*/
+		_backHistoryStack: [],
+
+		/**
+		* history.forward() and history.go() can fire popstate event
+		* as does history.back().
+		* However we should accept popstate event only came from history.back().
+		* When it is true, we will skip popstate event.
+		*
+		* @private
+		*/
+		_ignorePopState: false,
+
+		/**
+		* If pushstate() is called during popstate is in progress,
+		* we pushed it into this queue.
+		*
+		* @private
+		*/
+		_pushBackQueue: [],
+
+		/**
+		* Flag for popstate event is bubbling
+		*
+		* @private
+		*/
+		_isPopStateInProgress: false,
+
+		/**
+		* If "disableBackHistoryAPI" in AppInfo.json is set to true, this property
+		* should be false
+		*
+		* @type {Bollean}
+		* @default true
+		* @public
+		*/
+		enableBackHistoryAPI: true,
+
+		published: {
+			/**
+			* Flag for back key handler progress
+			*
+			* @public
+			*/
+			isBackInProgress: false
+		},
+
+		events: {
```

Comment:
Do we need to fire and handle the `onPopBackHistory` event or can we just directly call the `popBackHistory` method where needed?